File: programmers/kakao/level2/k진수에서_소수_개수_구하기.py
def nTok(n, k):
    tmp = ""
    while n:
        tmp = str(n%k) + tmp
        n //= k
    return tmp

def isPrime(x):
    if x <= 1: return False
    i = 2
    while i*i <= x:
        if x%i == 0: return False
        i += 1
    return True

def solution(n, k):
    answer = 0
    arr = nTok(n, k).split("0")
    for a in arr:
        if not a:
            continue
        elif isPrime(int(a)):
            answer += 1
        else:
            continue
    return answer


n, k = 437674, 3
print(solution(n, k)) # 3

n, k = 110011, 10
print(solution(n, k)) # 2

File: Java/programmers/kakao/level2/k진수에서소수개수구하기.java
package CodeTest.Java.programmers.kakao.level2;

public class k진수에서소수개수구하기 {
    public int solution(int n, int k) {
        int answer = 0;
        String[] arr = nTok(n, k).split("0");
        
        for (String a : arr) {
            if (!a.equals("") && isPrime(Long.parseLong(a))) answer++;
        }
        
        return answer;
    }
    
    private String nTok(int n, int k) {
        String tmp = "";
        while (n != 0) {
            tmp = String.valueOf(n%k) + tmp;
            n /= k;
        }
        return tmp;
    }
    
    private boolean isPrime(long x) {
        if (x <= 1) return false;
        long i = 2;
        while (i*i <= x) {
            if (x%i == 0) return false;
            i++;
        }
        return true;
    }
}
